src/postgres/tree/PostgresDatabaseTreeItem.ts

```diff
@@ -4,24 +4,44 @@
  *--------------------------------------------------------------------------------------------*/
 
 import { Client, ClientConfig } from 'pg';
-import pgStructure from 'pg-structure';
+import pgStructure, { Db } from 'pg-structure';
 import * as vscode from 'vscode';
-import { AzureParentTreeItem, ISubscriptionContext } from 'vscode-azureextensionui';
+import { AzExtTreeItem, AzureParentTreeItem, GenericTreeItem, IParsedError, ISubscriptionContext, parseError } from 'vscode-azureextensionui';
 import { getThemeAgnosticIconPath } from '../../constants';
+import { ext } from '../../extensionVariables';
+import { KeyTar, tryGetKeyTar } from '../../utils/keytar';
+import { localize } from '../../utils/localize';
 import { nonNullProp } from '../../utils/nonNull';
 import { PostgresSchemaTreeItem } from './PostgresSchemaTreeItem';
 import { PostgresServerTreeItem } from './PostgresServerTreeItem';
 
+interface IPersistedServer {
+    id: string;
+    username: string;
+}
+
 export class PostgresDatabaseTreeItem extends AzureParentTreeItem<ISubscriptionContext> {
     public static contextValue: string = "postgresDatabase";
     public readonly contextValue: string = PostgresDatabaseTreeItem.contextValue;
     public readonly childTypeLabel: string = "Schema";
     public readonly databaseName: string;
     public readonly parent: PostgresServerTreeItem;
 
+    private readonly _serviceName: string = "ms-azuretools.vscode-cosmosdb.postgresPasswords";
+    private _keytar: KeyTar | undefined;
+    private _usernameSuffix: string;
+    private _usernamePlaceholder: string;
+    private _usernameRegex: RegExp;
+    private _serverId: string;
+
     constructor(parent: PostgresServerTreeItem, databaseName: string) {
         super(parent);
         this.databaseName = databaseName;
+        this._keytar = tryGetKeyTar();
+        this._usernameSuffix = `@${this.parent.server.name}`;
```

Comment:
Probably don't need to make this a property this either since it's only used in the constructor.

Reply:
It's used to make the username regex too is it ok to keep it a property?

Reply:
Oh right, haha you got rid of the regex property.